### PR TITLE
Synkron-277 Added getItem, getItemById and it's needed POJOs

### PR DESCRIPTION
**Pull request by hapanivivek**
Added object for json data

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,178 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>io.zohoinventory</groupId>
+    <artifactId>zoho-inventory</artifactId>
+    <version>1.0.3</version>
```
Comment (manan-dft) [inline pom.xml]: Version should be 1.0.0

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+    <packaging>jar</packaging>
+
+    <name>zoho-inventory</name>
+    <description>zoho-inventory API using JDK 11</description>
+    <url>https://github.com/dftrakesh/zoho-Iinventory</url>
```
Comment (manan-dft) [inline pom.xml]: Verify this URL

File: src/main/java/io/zohoinventory/ZohoInventorySdk.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+import java.util.concurrent.CompletableFuture;
+
+import static io.zohoinventory.constantcode.ConstantCodes.*;
+
+@AllArgsConstructor
+@Builder(toBuilder = true)
+public class ZohoInventorySdk {
+
+    protected HttpClient client;
+    protected AccessCredentials accessCredential;
```
Comment (manan-dft) [inline src/main/java/io/zohoinventory/ZohoInventorySdk.java]: Add declaration in ascending order or length.

File: src/main/java/io/zohoinventory/constantcode/ConstantCodes.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
+package io.zohoinventory.constantcode;
+
+public interface ConstantCodes {
```
Comment (manan-dft) [inline src/main/java/io/zohoinventory/constantcode/ConstantCodes.java]: Remove codes which are not in use.

File: src/main/java/io/zohoinventory/models/authenticationapi/AccessCredentials.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+import java.time.LocalDateTime;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder(toBuilder = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)
```
Comment (manan-dft) [inline src/main/java/io/zohoinventory/models/authenticationapi/AccessCredentials.java]: All annotations are in use ?

**Review by manan-dft — CHANGES_REQUESTED** [review]
Check all my comments

File: pom.xml (hunk 1 above)
Comment (hapanivivek) [reply pom.xml]: Made changes.

File: pom.xml (hunk 2 above)
Comment (hapanivivek) [reply pom.xml]: made changes to inventory

File: src/main/java/io/zohoinventory/ZohoInventorySdk.java (hunk 3 above)
Comment (hapanivivek) [reply src/main/java/io/zohoinventory/ZohoInventorySdk.java]: made declaration in ascending order

File: src/main/java/io/zohoinventory/constantcode/ConstantCodes.java (hunk 4 above)
Comment (hapanivivek) [reply src/main/java/io/zohoinventory/constantcode/ConstantCodes.java]: all the constant are in use

File: src/main/java/io/zohoinventory/models/authenticationapi/AccessCredentials.java (hunk 5 above)
Comment (hapanivivek) [reply src/main/java/io/zohoinventory/models/authenticationapi/AccessCredentials.java]: all the annotations are in need for json data

File: src/main/java/io/zohoinventory/models/authenticationapi/AccessTokenResponse.java [hunk 6]
```diff
@@ -0,0 +1,19 @@
+package io.zohoinventory.models.authenticationapi;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.PropertyNamingStrategies;
+import com.fasterxml.jackson.databind.annotation.JsonNaming;
+import lombok.Data;
+
+@Data
+@JsonInclude(JsonInclude.Include.NON_NULL)
```
Comment (manan-dft) [inline src/main/java/io/zohoinventory/models/authenticationapi/AccessTokenResponse.java]: Remove it

File: src/main/java/io/zohoinventory/models/itemapi/Item.java [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+    private String attributeId1;
+    private String attributeId2;
+    private String attributeId3;
+    private String manufacturer;
+    private Boolean isReturnable;
+    private String dimensionUnit;
+    private Double taxPercentage;
+    private String zcrmProductId;
+    private Boolean hasAttachment;
+    private Date lastModifiedTime;
```
Comment (manan-dft) [inline src/main/java/io/zohoinventory/models/itemapi/Item.java]: Convert it into localdatetime

File: src/main/java/io/zohoinventory/ItemAPI.java [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+    public ItemAPI(AccessCredentials accessCredentials){
+        super(accessCredentials);
+    }
+
+    public ItemsWrapper getItems(HashMap<String, String> params)
+    {
+        refreshAccessToken();
+        URI uri = URI.create(API_BASED_END_POINT.concat(SLASH_CHARACTER)
+                .concat(VERSION)
+                .concat(SLASH_CHARACTER)
```
Comment (manan-dft) [inline src/main/java/io/zohoinventory/ItemAPI.java]: remove slash character from constatant codes

File: src/main/java/io/zohoinventory/ZohoInventorySdk.java [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+
+            HttpRequest request = HttpRequest.newBuilder(uriBuilder)
+                    .POST(HttpRequest.BodyPublishers.noBody())
+                    .header(CONTENT_TYPE, CONTENT_VALUE_APPLICATION_JSON)
+                    .build();
+
+            AccessTokenResponse accessTokenResponse = getRequestWrapped(request, AccessTokenResponse.class);
+            accessCredential.setAccessToken(accessTokenResponse.getAccessToken());
+            accessCredential.setExpiresInTime(LocalDateTime.now().plusSeconds(accessTokenResponse.getExpiresIn()));
+
```
Comment (manan-dft) [inline src/main/java/io/zohoinventory/ZohoInventorySdk.java]: remove extra line

File: src/main/java/io/zohoinventory/models/authenticationapi/AccessCredentials.java [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.PropertyNamingStrategies;
+import com.fasterxml.jackson.databind.annotation.JsonNaming;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+
+import java.time.LocalDateTime;
+
+@Data
+@AllArgsConstructor
```
Comment (manan-dft) [inline src/main/java/io/zohoinventory/models/authenticationapi/AccessCredentials.java]: remove unused annotation

File: src/main/java/io/zohoinventory/ItemAPI.java (hunk 8 above)
Comment (hapanivivek) [reply src/main/java/io/zohoinventory/ItemAPI.java]: Removed Slash Character

File: src/main/java/io/zohoinventory/ZohoInventorySdk.java (hunk 9 above)
Comment (hapanivivek) [reply src/main/java/io/zohoinventory/ZohoInventorySdk.java]: Removed Extra lines

File: src/main/java/io/zohoinventory/models/orderapi/Order.java [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+    private String shipmentDays;
+    private String dueByDays;
+    private String dueInDays;
+    private String currencyId;
+    private String source;
+    private String currencyCode;
+    private Double total;
+    private Double bcyTotal;
+    private Double totalInvoicedAmount;
+    @JsonSerialize(using = LocalDateTimeSerializer.class)
```
Comment (manan-dft) [inline src/main/java/io/zohoinventory/models/orderapi/Order.java]: leave one line above

File: src/main/java/io/zohoinventory/ItemAPI.java [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+
+public class ItemAPI extends ZohoInventorySdk{
+
+    public ItemAPI(AccessCredentials accessCredentials){
+        super(accessCredentials);
+    }
+
+    public ItemsWrapper getItems(HashMap<String, String> params) {
+        refreshAccessToken();
+        URI uri = URI.create(API_BASED_END_POINT
```
Comment (manan-dft) [inline src/main/java/io/zohoinventory/ItemAPI.java]: declare the variable String uri = API_BASED_END_POINT + VERSION + ITEM_ENDPOINT;
and pass it into URI.create

**Review by manan-dft — APPROVED** [review]

**Pull request closed without merging**